lib/core/src/Cardano/Wallet.hs

```diff
@@ -1235,377 +1212,182 @@ normalizeDelegationAddress s addr = do
-  where
-    db  = ctx ^. dbLayer @s @k
-    pid = PoolId (error "Dummy pool id for estimation. Never evaluated.")
+    let utxo :: UTxOIndex
+        utxo = UTxOIndex.fromUTxO $ availableUTxO @s pending cp
 
--- | Constructs a set of coin selections that select all funds from the given
---   source wallet, returning them as change.
---
--- If the coin selections returned by this function are used to create
--- transactions from the given wallet to a target wallet, executing those
--- transactions will have the effect of migrating all funds from the given
--- source wallet to the specified target wallet.
-selectCoinsForMigration
-    :: forall ctx s k n.
-        ( HasTransactionLayer k ctx
-        , HasLogger WalletLog ctx
-        , HasDBLayer s k ctx
-        , PaymentAddress n ByronKey
-        )
-    => ctx
-    -> WalletId
-       -- ^ The source wallet ID.
-    -> ExceptT ErrSelectForMigration IO ([CoinSelection], Coin)
-selectCoinsForMigration ctx wid = do
-    (utxo, _, txp, minUtxo) <- withExceptT ErrSelectForMigrationNoSuchWallet $
-        selectCoinsSetup @ctx @s @k ctx wid
-    selectCoinsForMigrationFromUTxO @ctx @k @n ctx utxo txp minUtxo wid
-
-selectCoinsForMigrationFromUTxO
-    :: forall ctx k n.
-        ( HasTransactionLayer k ctx
-        , HasLogger WalletLog ctx
-        , PaymentAddress n ByronKey
-        )
-    => ctx
-    -> W.UTxO
-    -> W.TxParameters
-    -> W.Coin
-    -> WalletId
-       -- ^ The source wallet ID.
-    -> ExceptT ErrSelectForMigration IO ([CoinSelection], Coin)
-selectCoinsForMigrationFromUTxO ctx utxo txp minUtxo wid = do
-    let feePolicy@(LinearFee (Quantity a) _) = txp ^. #getFeePolicy
-    let feeOptions = (feeOpts tl Nothing Nothing txp minBound mempty)
-            { estimateFee = minimumFee tl feePolicy Nothing Nothing . worstCase
-            , dustThreshold = max (Coin $ ceiling a) minUtxo
-            }
-    let selOptions = coinSelOpts tl (txp ^. #getTxMaxSize) Nothing
-    let previousDistribution = W.computeUtxoStatistics W.log10 utxo
-    liftIO $ traceWith tr $ MsgMigrationUTxOBefore previousDistribution
-    case depleteUTxO feeOptions (idealBatchSize selOptions) utxo of
-        cs | not (null cs) -> do
-            let resultDistribution = W.computeStatistics getCoins W.log10 cs
-            liftIO $ traceWith tr $ MsgMigrationUTxOAfter resultDistribution
-            liftIO $ traceWith tr $ MsgMigrationResult cs
-            let leftovers =
-                    unCoin (TokenBundle.getCoin $ W.balance utxo)
-                    -
-                    W.balance' (concatMap inputs cs)
-            pure (cs, Coin leftovers)
-        _ -> throwE (ErrSelectForMigrationEmptyWallet wid)
+    liftIO $ traceWith tr $ MsgSelectionStart utxo outs
+    sel <- performSelection
+        (calcMinimumCoinValue tl pp)
+        (calcMinimumCost tl pp tx)
+        (initSelectionCriteria tl pp tx utxo outs)
+    liftIO $ traceWith tr $ MsgSelectionDone sel
```

Comment:
Will this produce very large output in the log?

Reply:
Arf. Good point, because of the `utxoRemaining` in the `SelectionResult`, this can be quite large -- though only at debug severity. I'll tweak the message to text result to prevent this.

Reply:
The `Buildable` instance was actually already sound:

```hs
buildSelectionResult
    :: (NonEmpty change -> Builder)
    -> SelectionResult change
    -> Builder
buildSelectionResult changeF s@SelectionResult{inputsSelected,extraCoinSource} =
    mconcat
        [ nameF "inputs selected" (inputsF inputsSelected)
        , nameF "extra coin input" (build extraCoinSource)
        , nameF "outputs covered" (build $ outputsCovered s)
        , nameF "change generated" (changeF $ changeGenerated s)
        , nameF "size utxo remaining" (build $ UTxOIndex.size $ utxoRemaining s)
        ]
  where
    inputsF :: NonEmpty (TxIn, TxOut) -> Builder
    inputsF = blockListF' "+" tupleF
```

so, all good.